cli/src/pcluster/commands.py

```diff
@@ -569,9 +584,83 @@ def build_image(args):
             imagebuilder_config=load_yaml_dict(args.config_file), image_name=args.image_name, region=utils.get_region()
         )
         LOGGER.info("Response:")
-        LOGGER.info({"image": response.__repr__()})
+        if isinstance(response, ApiFailure):
+            LOGGER.info(f"Build image error {response.message}")
+        else:
+            LOGGER.info({"image": response.__repr__()})
     except Exception as e:
         utils.error(
             "Error parsing configuration file {0}.\nDouble check it's a valid Yaml file. "
             "Error: {1}".format(args.config_file, str(e))
         )
+
+
+def delete_image(args):
+    """Delete image described by image_name."""
+    LOGGER.info("Deleting: %s", args.image_name)
+    LOGGER.debug("CLI args: %s", str(args))
+    try:
+        # delete image raises an exception if stack does not exist
+        result = PclusterApi().delete_image(image_name=args.image_name, region=utils.get_region())
+        if isinstance(result, ImageInfo):
+            print(f"Image deletion started correctly. {result}")
+        else:
+            utils.error(f"Image deletion failed. {result.message}")
+
+        sys.stdout.write("\rStatus: %s" % result.stack_status)
+        sys.stdout.flush()
+        LOGGER.debug("Status: %s", result.stack_status)
+
+        if not args.nowait:
+            while result.stack_status == "DELETE_IN_PROGRESS":
+                time.sleep(1)
+                result = PclusterApi().describe_image(image_name=args.image_name, region=utils.get_region())
+                if isinstance(result, ImageInfo):
+                    events = utils.get_stack_events(result.stack_name, raise_on_error=True)[0]
+                    resource_status = (
+                        "Status: %s - %s" % (events.get("LogicalResourceId"), events.get("ResourceStatus"))
+                    ).ljust(80)
+                    sys.stdout.write("\r%s" % resource_status)
+                    sys.stdout.flush()
+                elif isinstance(result, ApiFailure):
+                    # If stack is already deleted
+                    if f"ImageBuilder stack {args.image_name} does not exist." in result.message:
+                        LOGGER.info("\nImage deleted successfully.")
```

Comment:
LOGGER.warning(f"\nImage {args.image_name} has already been deleted or does not exist.")

Reply:
Done